sqlglot/optimizer/qualify_columns.py

```diff
@@ -297,9 +324,20 @@ def _expand_stars(scope, resolver, using_column_tables):
         for table in tables:
             if table not in scope.sources:
                 raise OptimizeError(f"Unknown table: {table}")
+
             columns = resolver.get_source_columns(table, only_visible=True)
 
             if columns and "*" not in columns:
+                if has_pivoted_source:
+                    implicit_columns = list(set(columns) - pivot_columns)
+                    new_selections.extend(
+                        [
```

Comment:
can you make this a generator instead ? ( instead of [

Reply:
Sure